tests/conftest.py

```diff
@@ -225,25 +228,29 @@ def pytest_runtest_setup(item: "pytest.Item"):
             pytest.skip("Skipping old migration test")
 
 
+# adds a custom log location for postgres
+postgresql_proc_with_log = factories.postgresql_proc(startparams=f"--log={pg_logfile}")
+
+
 @pytest.fixture(scope="session")
 def postgres_db(request: pytest.FixtureRequest):
     """This fixture loads the pytest-postgresql fixture. When --postgresql-host is set, it will use the noproc
     fixture to use an external database. Without this option, an "embedded" postgres is started.
     """
+
     option_name = "postgresql_host"
     conf = request.config.getoption(option_name)
     if conf:
         fixture = "postgresql_noproc"
     else:
-        fixture = "postgresql_proc"
+        fixture = "postgresql_proc_with_log"
 
     logger.info("Using database fixture %s", fixture)
     pg = request.getfixturevalue(fixture)
     yield pg
 
-    logfile = os.path.join(initial_cwd, "pg.log")
-    if os.path.exists(logfile):
-        with open(logfile, "r") as fh:
+    if os.path.exists(pg_logfile):
```

Comment:
The log file is not deleted at the end of the test suite. Is this something we want? Also, new logs are appended to the old log file. So if we don't clean up at the end of the test suite, we at least need to truncate the file at the start of the test case.

Reply:
I added a  os.remove(pg_logfile) if the file exists

Reply:
The log file is not deleted when there is a deadlock. This could be a feature or a bug. In any case, if it's a feature we should truncate the file the next time the test is run.